src/modules/processing/services/tiktok.js

```diff
@@ -19,54 +20,55 @@ export default async function(obj) {
 
         if (!html) return { error: 'ErrorCouldntFetch' };
 
-        if (html.slice(0, 17) === '<a href="https://') {
-            postId = html.split('<a href="https://')[1].split('?')[0].split('/')[3]
-        } else if (html.slice(0, 32) === '<a href="https://m.tiktok.com/v/' && html.includes('/v/')) {
-            postId = html.split('/v/')[1].split('.html')[0].replace("/", '')
+        if (html.startsWith('<a href="https://')) {
+            const { patternMatch } = extract(html.split('<a href="https://')[1].split('?')[0])
+            username = patternMatch.user
+            postId = patternMatch.postId
         }
     }
-    if (!postId) return { error: 'ErrorCantGetID' };
-
-    let deviceInfo = new URLSearchParams(env.tiktokDeviceInfo).toString();
+    if (!username || !postId) return { error: 'ErrorCantGetID' };
 
-    let apiURL = new URL(apiPath);
-        apiURL.searchParams.append("aweme_id", postId);
-
-    let detail = await fetch(`${apiURL.href}&${deviceInfo}`, {
+    // should always be /video/, even for photos
+    const res = await fetch(`https://tiktok.com/${username}/video/${postId}`, {
```

Comment:
username isn't needed for getting the video page, using `i` instead of username works fine. please remove all references to the username across files, as it's redundant.